assets/js/components/display-setting.test.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * DisplaySetting tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import DisplaySetting from './display-setting';
+import { render } from '../../../tests/js/test-utils';
+
+describe( 'DisplaySetting', () => {
+	it( 'returns non-empty string unchanged', () => {
+		const { container } = render( <DisplaySetting value="Test Value" /> );
+		expect( container.textContent ).toBe( 'Test Value' );
```

Comment:
React Testing Library uses [`jest-dom`](https://github.com/testing-library/jest-dom) so we can actually use a more expressive matcher for this like so:
```suggestion
		expect( container ).toHaveTextContent( 'Test Value' );
```

Would you please update these tests to use this instead?

Reply:
Yeah, that's the method I tried originally, but `toHaveTextContent()` either normalizes white space or replaces `&nbsp;`s (see reference below) which prevents us from being able to test for that `&nbsp;`. Testing with `textContent` was a workaround. We could use `toHaveTextContent()` for the first case with the non-empty string though. Let me know if I should change that or keep them consistent within this file. (Or if there's a preferred workaround).

[https://github.com/testing-library/jest-dom/blob/d6655dbd0736f9ab8391b59570a0ecd631c93916/src/to-have-text-content.js#L11-L13](https://github.com/testing-library/jest-dom/blob/d6655dbd0736f9ab8391b59570a0ecd631c93916/src/to-have-text-content.js#L11-L13)

Reply:
That makes sense @jqlee85. I see there is an option to turn that off with `{ normalizeWhitespace: false }` as the second parameter to the matcher, but that's a bit overly verbose for each I think.

On its face, I like your approach better as that's still quite clear and is less verbose but the text content-specific matcher gives us more specific feedback in the event of a failure.

Let's go with the `toHaveTextContent()` matcher for the better feedback (unless there's another reason why we can't), but also for consistency with how we make assertions on text content. We can DRY up the usage with a variable for `options` above that they can all share. Maybe this variable could even be called `preserveWhitespace` to be clear about what it's doing?

If for some reason we still can't use this matcher, then let's just document the exception with a comment that explains why 😄

Reply:
@aaemnnosttv the `{normalizeWhitespace: false}` option doesn't resolve the issue, unfortunately. It just causes toHaveTextContent() to explicitly replace **nbsp**s with spaces:

`htmlElement.textContent.replace(/\u00a0/g, ' ') // Replace &nbsp; with normal spaces`.

It's as if toHaveTextContent() was specifically designed to break this test :laughing:.

I've updated the test to use toHaveTextContent for the first case, and added comments to the rest to document why we're using toBe instead.